### PR TITLE
feat(ui): download installation output

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add downloading of installation output to the logs tab.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View a new machine.
- Go to the React logs tab (change /summary in the URL to /logs).
- There should be no "Installation output" in the download selector.
- Go to the React logs tab of a deployed machine.
- There should be an "Installation output" download. Click it and it should download a file containing the log.

## Fixes

Fixes: canonical-web-and-design/maas-squad#2369.

### CODE REVIEW

File: ui/src/app/machines/views/MachineDetails/MachineLogs/MachineLogs.tsx [hunk 1]
```diff
@@ -48,11 +48,8 @@ const MachineNetwork = ({ systemId }: Props): JSX.Element => {
             },
           ]}
         />
-        <DownloadMenu />
+        <DownloadMenu systemId={systemId} />
       </div>
-      <Route exact path="/machine/:id/logs/events">
```
Comment (huwshimi) [inline ui/src/app/machines/views/MachineDetails/MachineLogs/MachineLogs.tsx:53]: A drive-by fix for a rebase that caused duplicate event components.

File: ui/src/app/machines/views/MachineDetails/MachineLogs/DownloadMenu/DownloadMenu.tsx [hunk 2]
```diff
@@ -16,9 +22,21 @@ export const DownloadMenu = (): JSX.Element => {
...
-          },
+          ...(installationOutput.log
+            ? [
+                {
+                  children: "Installation output",
+                  onClick: () => {
+                    if (installationOutput.log) {
+                      fileDownload(
+                        installationOutput.log,
+                        "installation-output.txt"
```
Comment (anthonydillon) [inline ui/src/app/machines/views/MachineDetails/MachineLogs/DownloadMenu/DownloadMenu.tsx:33]: Could we help the user out by naming the file after the machine or something?

Comment (anthonydillon) [reply ui/src/app/machines/views/MachineDetails/MachineLogs/DownloadMenu/DownloadMenu.tsx:33]: Maybe even a date on it?

**Review by anthonydillon — APPROVED** [review]
Nice work, this is great. One small suggestion in the code.

File: ui/src/app/machines/views/MachineDetails/MachineLogs/DownloadMenu/DownloadMenu.tsx (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/machines/views/MachineDetails/MachineLogs/DownloadMenu/DownloadMenu.tsx:33]: Good idea!
